File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        
        int zeroCount=0;
        int transCount=0;
        while(!s.equals("1")){
            for(int i=0; i<s.length(); i++){        // 제거할 0의 개수
                if(s.charAt(i) == '0'){
                    zeroCount++;
                }
            }
            s = s.replace("0", "");     // 0 제거
            
            s = Integer.toBinaryString(s.length());     // 제거 후 길이를 이진수로 변환
            transCount++;
        }
        
        int[] answer = {transCount, zeroCount};
        return answer;
    }
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = []
    
    zeros = 0
    counts = 0
    
    while s != '1':
        zeros += s.count('0')
        print(zeros)
        counts += 1
        s = s.replace('0', '')
        s = format(len(s), 'b')       # 접두어(0b) 제외
        
    answer.append(counts)
    answer.append(zeros)
    
    return answer